app/assets/javascripts/countdown.js

```diff
@@ -0,0 +1,38 @@
+function Countdown(trackingElements) {
+  this.trackingElements = trackingElements;
+  this.auction_end_time = trackingElements.auction_end_time.text();
+
+  this.ongoing = function(durationCalculator) {
+    this.trackingElements.days.html(durationCalculator.days);
+    this.trackingElements.hours.html(durationCalculator.hours);
+    this.trackingElements.minutes.html(durationCalculator.minutes);
+    this.trackingElements.seconds.html(durationCalculator.seconds);
+  };
+
+  this.onEnd = function(durationCalculator) {
+    this.trackingElements.auction_end_time.html("Auction for this item is closed");
+    this.trackingElements.buy_now_button.hide();
+    this.trackingElements.bid_now_form.hide();
+  };
+
+  this.start = function() {
+    calc = new DurationCalculator(this.auction_end_time);
+    duration = calc.duration;
+    if(duration >= 0){
+      this.ongoing(calc);
```

Comment:
This `ongoing` name doesn't really make sense. Seems like it just "updatesElements" or something.